frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -1385,6 +1385,15 @@ QualifiedType getInstantiationType(Context* context,
       // now construct the ClassType
       auto ct = ClassType::get(context, bct, manager, dec);
       return QualifiedType(formalType.kind(), ct);
+    } else if (auto formalRT = formalT->toRecordType()) {
+      if (formalRT->name() == UniqueString::get(context, "_owned") ||
+          formalRT->name() == UniqueString::get(context, "_shared")) {
+        auto bct = actualCt->basicClassType();
+        auto manager = actualCt->manager();
+        auto dec = actualCt->decorator();
+        auto ct = ClassType::get(context, bct, manager, dec);
```

Comment:
Is this not re-constructing `actualCt` in its entirety? Seems like you can just return `actualCt` -- but I think it's possible for `_owned` to be itself instantiated, so maybe this code is not quite sufficient anyway.